### ISSUE TITLE
no-this-assignment false positive

### ISSUE THREAD

**Issue by mihai-dinculescu** [palantir/tslint#3002]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const { props } = this;
```

#### Actual behavior
severity: 'Error'
message: 'Don't assign members of `this` to local variables. (no-this-assignment)'

#### Expected behavior
no error

**Comment by adidahiya** [comment]
this is mentioned in the docs. you can allow that code by using the `allow-destructuring` option. https://palantir.github.io/tslint/rules/no-this-assignment/
